pandas/core/generic.py

```diff
@@ -4079,6 +4105,149 @@ def _reindex_axis(self, new_index, fill_method, axis, copy):
+
+        Returns
+        -------
+        Nothing, the object is modified inplace.
+
+        See Also
+        --------
+        Series.update : Similar method for `Series`.
+        DataFrame.merge : For column(s)-on-columns(s) operations.
+        dict.update : Similar method for `dict`.
+
+        Examples
+        --------
+        >>> df = pd.DataFrame({'A': [1, 2, 3],
+        ...                    'B': [400, 500, 600]})
+        >>> new_df = pd.DataFrame({'B': [4, 5, 6],
+        ...                        'C': [7, 8, 9]})
+        >>> df.update(new_df)
+        >>> df
+           A  B
+        0  1  4
+        1  2  5
+        2  3  6
+
+        The DataFrame's length does not increase as a result of the update,
+        only values at matching index/column labels are updated.
+
+        >>> df = pd.DataFrame({'A': ['a', 'b', 'c'],
+        ...                    'B': ['x', 'y', 'z']})
+        >>> new_df = pd.DataFrame({'B': ['d', 'e', 'f', 'g', 'h', 'i']})
+        >>> df.update(new_df)
+        >>> df
+           A  B
+        0  a  d
+        1  b  e
+        2  c  f
+
+        For Series, it's name attribute must be set.
+
+        >>> df = pd.DataFrame({'A': ['a', 'b', 'c'],
+        ...                    'B': ['x', 'y', 'z']})
+        >>> new_column = pd.Series(['d', 'e'], name='B', index=[0, 2])
+        >>> df.update(new_column)
+        >>> df
+           A  B
+        0  a  d
+        1  b  y
+        2  c  e
+        >>> df = pd.DataFrame({'A': ['a', 'b', 'c'],
+        ...                    'B': ['x', 'y', 'z']})
+        >>> new_df = pd.DataFrame({'B': ['d', 'e']}, index=[1, 2])
+        >>> df.update(new_df)
+        >>> df
+           A  B
+        0  a  x
+        1  b  d
+        2  c  e
+
+        If `other` contains NaNs the corresponding values are not updated
+        in the original dataframe.
+
+        >>> df = pd.DataFrame({'A': [1, 2, 3],
+        ...                    'B': [400, 500, 600]})
+        >>> new_df = pd.DataFrame({'B': [4, np.nan, 6]})
+        >>> df.update(new_df)
+        >>> df
+           A      B
+        0  1    4.0
+        1  2  500.0
+        2  3    6.0
+        """
+        from pandas import Series, DataFrame
+        # TODO: Support other joins
+        if join != 'left':  # pragma: no cover
+            raise NotImplementedError("Only left join is supported")
+
+        if isinstance(self, ABCSeries):
+            if not isinstance(other, ABCSeries):
+                other = Series(other)
+            other = other.reindex_like(self)
+            this = self.values
+            that = other.values
+            updated = _update_column(this, that, overwrite=overwrite,
+                                     filter_func=filter_func,
+                                     raise_conflict=raise_conflict)
+            if updated is None:
+                # don't overwrite Series unnecessarily
+                return
+            self._data._block.values = updated
```

Comment:
this is SO unsafe, doesn't handle caching and is reaching into internals. Use ``self._update_inplace(updated)``

Reply:
```
if updated is not None:
    self._update_inplace(updated)
```